python/lsst/dax/distribution/DataGenClient.py

```diff
@@ -0,0 +1,786 @@
+                        if not foundCsv:
+                            print("ERROR Problems with finding essential csv for creating overlap chunk=",
+                                  chunk, filesCsv)
+                            continue
+                        # Create edgeOnly neededChunks
+                        createdAllNeeded = True
+                        for nCh in neededChunks:
+                            genResult = self._generateChunk(nCh, edge_only=True)
+                            if genResult == 'failed':
+                                print("ERROR Failed to generate chunk", nCh)
+                                createdAllNeeded = False
+                                continue
+                        if createdAllNeeded:
+                            print("Created all needed edgeOnly for ", nCh)
+                            # Put hardlinks to all the files needed for a chunk in
+                            # a specific directory for the partioner to use to create
+                            # the overlap tables and so on.
+                            if self._fillChunkDir(chunk, neighborChunks):
+                                haveAllCsvChunks.append(chunk)
+                    # Generate overlap tables and files for ingest.
+                    # Start the transaction
+                    abort = False
+                    if len(haveAllCsvChunks):
+                        self._startTransaction()
+                    try:
+                        for chunk in haveAllCsvChunks:
+                            if self._createOverlapTables(chunk):
+                                print("created overlap for chunk", chunk)
+                                withOverlapChunks.append(chunk)
+                            else:
+                                print("ERROR failed to create overlap for chunk", chunk)
+                    except Exception as exc:
+                        # Abort the transaction if possible.
+                        print("ERROR transaction failed ", exc)
+                        abort = True
+                    # TODO: It may be better to have the server end the transaction as
+                    # it could give a better record of which chunks got ingested.
+                    success, status, content = self._endTransaction(abort)
+                    if success and not abort:
+                        for chunk in withOverlapChunks:
+                            ingestedChunks.append(chunk)
+                    else:
+                        print("Failed ingest, transaction failed or aborted ", status, content)
+                    # If no chunks were created, likely fatal error. Asking for more
+                    # chunks to create would just cause more problems.
+                    if len(ingestedChunks) == 0:
+                        print("ERROR no chunks were successfully ingested, ending program")
```

Comment:
It's better to tell the serve that nothing was created/ingested, otherwise the problems will register as a network failure instead of something wrong with this worker.